substate-store-impls/src/hash_tree/mod.rs

```diff
@@ -46,27 +97,31 @@ pub fn put_at_next_version<S: TreeStore>(
     node_tier_store: &S,
     node_root_version: Option<Version>,
     database_updates: &DatabaseUpdates,
+    store_values: bool,
```

Comment:
> When integrating with this in Node, I noticed that we actually have one other way of applying "disable storing values" config (namely, by 3 lines of code at our storage layer; honestly: easier than passing our config down to this method!).

I don't understand how the proposed 3 liner would work? Would it hackily insert values in after-the-fact or something?

Reply:
> I don't understand how the proposed 3 liner would work?

I deliberately said _one other way of applying "**disable** storing values" config_ :D

There is this existing tree-Node-saving loop inside our `fn commit(&self, commit_bundle: CommitBundle) {`:
```
for (key, node) in state_hash_tree_update.new_nodes {

  // The 3-liner:
  if !config.store_values_in_jmt {
    node = node.without_substate_value();
  }

  ... // yup, now save it
}
```

Reply:
In other words:
- JMT, as known by the Engine, _always_ stores historical values at its Substate-Tier leaf nodes.
  - in the code, that field is actually `Option<>` due to technical reasons (related to all 3 Tiers living in a single `TreeStore`), but in practice it will be present for every Substate-Tier leaf.
  - the above ugliness can be removed in another refactor (not trivial to do it right now).
- Node can post-process those nodes before actually storing them.

_(I love the name-clashing Node vs node :D)_